sdk/platform/http_client/curl/src/az_curl.c

```diff
@@ -588,14 +593,13 @@ static AZ_NODISCARD az_result _az_http_client_curl_send_request_impl_process(
   // make sure to set the end of the body response as the end of the complete response
   if (az_succeeded(result))
   {
-    az_span remaining
-        = az_span_slice_to_end(response->_internal.http_response, response->_internal.written);
-
-    AZ_RETURN_IF_NOT_ENOUGH_SIZE(remaining, 1);
-
-    az_span_copy_u8(remaining, 0);
-    response->_internal.written++;
+    az_result append_zero_to_respose = az_http_response_write_u8(response, 0);
+    if (az_failed(append_zero_to_respose))
```

Comment:
nit: can the method call be moved into the az_failed call?
```suggestion
    if (az_failed(az_http_response_write_u8(response, 0)))
```

Reply:
not, we need to return that error if it failed.